make/tests

```diff
@@ -1,67 +1,46 @@
 N_TESTS ?= 100 # override this in make/local. If <= 0, N_TESTS + 1 is interpreted as the number of batches to group the tests into
 
 ##
-# LIBGTEST is the google test library
 # GTEST_MAIN is the file that contains the google test
 ##
-LIBGTEST = test/libgtest.a
 GTEST_MAIN = $(GTEST)/src/gtest_main.cc
-CFLAGS_GTEST += -isystem $(GTEST)/include -isystem $(GTEST)
+GTEST_CXXFLAGS += -isystem $(GTEST)/include -isystem $(GTEST) -O$O
 
 ##
 # Build the google test library.
 ##
-$(LIBGTEST): $(LIBGTEST)(test/gtest.o)
-
-test/gtest.o: $(GTEST)/src/gtest-all.cc
-	@mkdir -p test
-	$(COMPILE.c) -O$O $(CFLAGS_GTEST) $< $(OUTPUT_OPTION)
-
-##
-# Rule for building a test
-##
-test/%_test.o : test/%_test.cpp
-	@mkdir -p $(dir $@)
-	$(COMPILE.c) -O$O $(CFLAGS_GTEST) $< $(OUTPUT_OPTION)
-
-##
-# Rule for building a test executable
-##
-test/%$(EXE) : test/%.o $(LIBGTEST) $(LIBCVODES)
-	@mkdir -p $(dir $@)
-	$(LINK.c) -O$O $(GTEST_MAIN) $< $(CFLAGS_GTEST) $(OUTPUT_OPTION) $(LIBGTEST) $(LDLIBS) $(LIBCVODES)
+$(GTEST)/src/gtest-all.o: CXXFLAGS += $(GTEST_CXXFLAGS)
 
+test/%$(EXE) : CXXFLAGS += $(GTEST_CXXFLAGS)
+test/%$(EXE) : test/%.o $(GTEST_MAIN) $(GTEST)/src/gtest-all.o
+	$(LINK.cpp) $^ $(LOADLIBES) $(LDLIBS) -o $@
 
 ##
 # Rule for generating dependencies.
 ##
-.PRECIOUS: test/%_test.d
-test/%_test.d : test/%_test.cpp
+.PRECIOUS: test/%.d
+test/%.d : test/%.cpp
 	@mkdir -p $(dir $@)
 	@set -e; \
 	rm -f $@; \
-	$(CC) $(CFLAGS) -O$O $(CFLAGS_GTEST) $(TARGET_ARCH) -MM $< > $@.$$$$; \
-	sed -e 's,\($(notdir $*)\)\(_test\)\.o[ :]*,$(dir $@)\1_test\$(EXE) $@ : ,g' < $@.$$$$ > $@; \
+	$(COMPILE.cpp) -MM $< > $@.$$$$; \
```

Comment:
From what I read, `COMPILE.c` is for C and `COMPILE.cc` is for C++, and some versions don't have `COMPILE.cpp`: http://docs.oracle.com/cd/E19504-01/802-5880/6i9k05dhg/index.html